depthai_ros_driver/include/depthai_ros_driver/impl/depthai_base.hpp

```diff
@@ -273,123 +295,166 @@ void DepthAIBase<Node>::onInit() {
 
     get_param(nh, bool{}, "depthai_block_read", _depthai_block_read);
     get_param(nh, bool{}, "enable_sync", _sync_video_meta);
-    get_param(nh, bool{}, "full_fov_nn", _full_fov_nn);
-    get_param(nh, bool{}, "disable_depth", _compute_bbox_depth);
     get_param(nh, bool{}, "force_usb2", _force_usb2);
 
     get_param(nh, int{}, "rgb_height", _rgb_height);
     get_param(nh, int{}, "rgb_fps", _rgb_fps);
     get_param(nh, int{}, "depth_height", _depth_height);
     get_param(nh, int{}, "depth_fps", _depth_fps);
-    get_param(nh, int{}, "shaves", _shaves);
-    get_param(nh, int{}, "cmx_slices", _cmx_slices);
-    get_param(nh, int{}, "nn_engines", _nn_engines);
+
+    get_param(nh, bool{}, "extended_disparity", _extended_disparity);
+    get_param(nh, bool{}, "subpixel", _subpixel);
+    get_param(nh, bool{}, "lrcheck", _lrcheck);
 
     get_param(nh, int{}, "queue_size", _queue_size);
 
     if (_camera_param_uri.back() != '/') {
         _camera_param_uri += "/";
     }
 
+    // set unsupported flags
+    _request_jpegout = false;
+
+    // disparity mode
+    _maxDisp = 96;
+    if (_extended_disparity) _maxDisp *= 2;
+    if (_subpixel) _maxDisp *= 32; // 5 bits fractional disparity
+
+    // generate config json
+    _pipeline_config_json = generatePipelineConfigJson();
+
+    // Prepare streams (pub, sub, etc)
     prepareStreamConfig();
 
+
+    // Start pipeline
+    _pipeline_loader = std::make_unique<pluginlib::ClassLoader<rr::Pipeline>>("depthai_ros_driver", "rr::Pipeline");
+    try {
+        _pipeline_plugin = _pipeline_loader->createInstance(_pipeline_name);
+        _pipeline_plugin->configure(_pipeline_config_json);
+        _pipeline = _pipeline_plugin->getPipeline();
```

Comment:
Let's call initialize before calling `getPipeline`

Reply:
`configure` is working as an initialization.

Reply:
Oh, my bad.